File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        int indiceDoisPontos = logLine.indexOf(":");
        String mensagemNaoFormatada = logLine.substring(indiceDoisPontos + 1);

        return mensagemNaoFormatada.strip();
    }

    public static String logLevel(String logLine) {
        int indiceAbreColchetes = logLine.indexOf("[");
        int indiceFechaColchetes = logLine.indexOf("]");
        String nivelNaoFormatado = logLine.substring(indiceAbreColchetes + 1, indiceFechaColchetes);
        String nivelFormatado = nivelNaoFormatado.toLowerCase();

        return nivelFormatado;
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }

}

File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int levelIndex = logLine.IndexOf("]:") + 2;
        string subLog = logLine.Substring(levelIndex);
        return subLog.Trim();
    }

    public static string LogLevel(string logLine)
    {
        int beginIndex = logLine.IndexOf("[") + 1;
        int endIndex = logLine.IndexOf("]");
        string level = logLine.Substring(beginIndex, (endIndex-beginIndex));
        return level.ToLower();
    }

    public static string Reformat(string logLine)
    {
        string level = LogLevel(logLine);
        string message = Message(logLine);
        return $"{message} ({level})";
    }
}
